event_email_reminder/models/event.py

```diff
@@ -42,27 +41,26 @@ def run_event_email_reminder(
           want to notify.
         """
         today = fields.Date.context_today(self)
-        limit_date = fields.Date.to_string(
-            fields.Date.from_string(today) + timedelta(days=days),
-        )
+        time_start, time_end = time(0, 0, 0), time(23, 59, 59)
+        limit_date = today + timedelta(days=days)
         if draft_events:
             domain = [('state', 'in', ['draft', 'confirm'])]
         else:
             domain = [('state', '=', 'confirm')]
         if not near_events:
             domain.extend([
-                ('date_begin', '>=', '%s 00:00:00' % limit_date),
-                ('date_begin', '<=', '%s 23:59:59' % limit_date),
+                ('date_begin', '>=', datetime.combine(limit_date, time_start)),
```

Comment:
AFAIK, domains still use string format, so I think this is working by chance as the `repr` of the datetime matches the format, but that can be assured. You should do `fields.Datetime.to_string(...`

Reply:
I was curious. Actually it's the other way around: [if you use a string, it's converted to a datetime object. If you use a datetime, it's used raw](https://github.com/odoo/odoo/blob/9617e79e583aeb3d6d983045c5934dae28dce0c1/odoo/osv/expression.py#L1101-L1115).

It's interesting, though, that I could have saved the `00:00:00` and `23:59:59` thingies. :thinking:

Reply:
Well, seeing the code, it seems both are equally accepted, so good for that "polymorphism".